app/code/Magento/Directory/Helper/Data.php

```diff
@@ -406,14 +406,21 @@ public function getWeightUnit()
      * Get current scope from request
      *
      * @return array
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
      */
     private function getCurrentScope(): array
```

Comment:
This method has as a goal (as described on the PHP doc block) to get the current scope from the request, I'm not sure that such change makes sense, maybe we should look at the problem from the request perspective?

Reply:
Method getCurrentScope has been added in commit 2b4cfab, ticket MC-33168, and this change produced issue #30577. tbh, my change looks like quick fix. I am not sure that method getCurrentScope should be here. In this class we have store resolver. I think make sense to move logic of current scope determination into store resolver. But I don't know what issue MC-33168 was about, and afraid to broke something, what has been fixed previously. Therefore I decided change default value in method getCurrentScope. If somebody can explain me issue of MC-33168, then I'll be able to move on.

Reply:
I agree that `getCurrentScope` probably shouldn't be on this helper, but by adding that condition it makes the function ambiguous. My proposal would be to look at the solution from the Create Order perspective, see from which request this function is being called, and make sure we have the correct scope on the request, it should be also a quick fix, what do you think?